api/envoy/extensions/filters/listener/proxy_protocol/v3/proxy_protocol.proto

```diff
@@ -41,4 +41,20 @@ message ProxyProtocol {
 
   // The list of rules to apply to requests.
   repeated Rule rules = 1;
+
+  // Allow requests through that don't use proxy protocol. Defaults to false.
+  //
+  // .. attention::
+  //
+  //   This breaks conformance with the specification.
+  //   Only enable if ALL traffic to the listener comes from a trusted source.
+  //   For more information on the security implications of this feature, see
+  //   https://www.haproxy.org/download/2.1/doc/proxy-protocol.txt
+  //
+  //   While incredibly rare, requests of less than 12 bytes that match the
+  //   beginning of the proxy protocol v2 format and requests of less than 6
+  //   bytes that match the beginning of proxy protocol v1 format will timeout.
+  //   (envoy is unable to differentiate these requests from incomplete proxy
+  //   protocol requests)
```

Comment:
Also worth to mention if you want to continue the connection, set the `continue_on_listener_filters_timeout` as true.

Reply:
I thought the docs explicitly recommend not to do this? https://github.com/envoyproxy/envoy/blob/9d5627a0879b0a029e90515137c108e1d2884bfc/api/envoy/api/v2/listener.proto#L145-L152

Reply:
yes, we needn't add that, I'm thinking of the wrong case. Please ignore it.